app/questionnaire/rules/rule_evaluator.py

```diff
@@ -30,13 +29,13 @@
 @dataclass
 class RuleEvaluator:
     schema: QuestionnaireSchema
-    answer_store: AnswerStore
-    list_store: ListStore
+    answer_store: answer_store.AnswerStore
+    list_store: list_store.ListStore
```

Comment:
Why has this been done?
I think it would be good here to remove these parameters and replace them with `data_stores` then the placeholder renderer below can come from the data stores

Reply:
Circular import error.

Reply:
Same as other, ignore about replacing args, I think the imports can be fixed with `TYPE_CHECKED` - I see what you mean